CMake/docs.cmake

```diff
@@ -0,0 +1,22 @@
+option(BUILD_DOC "Build documentation" ON)
+
+# check if Doxygen is installed
+find_package(Doxygen)
```

Comment:
We have #421 for this, please coordinate there that you are implementing it here